ee/localserver/server.go

```diff
@@ -407,3 +410,93 @@ func (ls *localServer) rateLimitHandler(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r)
 	})
 }
+
+// munemoCheckHandler returns an http error if the request's X-Kolide-Munemo header does not match the organization in the enroll secret
+// if either the header or the enroll secret are missing, the request is allowed through
+func (ls *localServer) munemoCheckHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		munemoHeaderValue := r.Header.Get("X-Kolide-Munemo")
+		if munemoHeaderValue == "" {
+			ls.slogger.Log(r.Context(), slog.LevelDebug,
+				"munemo header is empty, continuing",
+			)
+
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		localServerMunemo, err := ls.getMunemoFromEnrollSecret()
+		if err != nil {
+			ls.slogger.Log(r.Context(), slog.LevelError,
+				"getting munemo from enroll secret, conitnuing",
+				"err", err,
+			)
+
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		if localServerMunemo == "" {
+			ls.slogger.Log(r.Context(), slog.LevelDebug,
+				"munemo in enroll secret is empty, continuing",
+				"header", munemoHeaderValue,
+			)
+
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		if munemoHeaderValue == localServerMunemo {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		ls.slogger.Log(r.Context(), slog.LevelDebug,
+			"munemo in request does not match munemo in enroll secret",
+			"header", munemoHeaderValue,
+			"enroll_secret", localServerMunemo,
+		)
+
+		http.Error(w, "munemo in request does not match munemo in enroll secret", http.StatusUnauthorized)
+		return
+	})
+}
+
+// munemoCheckHandler returns an http error if the request's X-Kolide-Munemo header does not match the organization in the enroll secret
```

Comment:
should probably update or remove this comment

Reply:
updated